GAEN-645, checking for phone numbers used every 5 minutes in callbac…

lundjrl opened the pull request:
…k form

#### Why:

Users can repeatedly spam the callback form with their phone number, this solution makes a timestamp on the first submit and saves the phone number. Only after 15 minutes can that phone number be reused on the same device.

#### This commit:

<!-- Describe how this PR achieves the desired change in functionality. -->

#### Screenshots:

<!-- Provide screenshots or gif for visual changes -->

#### How to test:

<!-- Provide steps to test this PR -->

#### Linked issues:

<!-- Add issues here e.g.: Fixes #1234 -->


---

<!-- Commits, PRs, and Code Review should follow these guidelines: -->

<!-- How to Write a Git Commit Message -->
<!-- https://chris.beams.io/posts/git-commit/ -->

<!-- The anatomy of a perfect pull request -->
<!-- https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067 -->

<!-- Implementing a Strong Code-Review Culture -->
<!-- https://www.youtube.com/watch?v=PJjmw9TRB7s -->

lundjrl committed:
# This is a combination of 4 commits.
# This is the 1st commit message:

GAEN-645, stop users from blasting the callback form over and over again.

# This is the commit message #2:

Removed space typo in common.hascode
# This is the commit message #3:

Removed need to save phone number in local storage and replaced with count instead

# This is the commit message #4:

Removed need to save phone number in local storage and replaced with count instead

lundjrl commented:
Note: This build fails one of the "describe" jest tests in Form.spec.tsx

artgibson commented on src/Callback/Form.tsx:
```diff
@@ -74,33 +75,84 @@ const CallbackForm: FunctionComponent = () => {
+  const initWaitForResubmit = async (phoneNumber: string) => {
+    const now = new Date()
+    const nowStr = now.getTime().toString()
+    await AsyncStorage.setItem("LastPhoneNumber", phoneNumber)
```
@lundjrl let's not store the phone number, cap the number of successful attempts to 2 per 15 minutes. Update error messages to handle that also.

lundjrl replied:
@artgibson  This should work without storing the phone number in local storage. We just check to see if the user has completed two submissions in the past 15 minutes and throw an error if so.

lundjrl replied:
@artgibson  can you approve so we can merge into develop?

artgibson commented on src/Callback/Form.tsx:
```diff
@@ -74,33 +75,78 @@ const CallbackForm: FunctionComponent = () => {
     }
   }
 
+  /* Wait for 15 minutes to pass before submitting another phonenumber after two have been submitted. */
```
@lundjrl clean up this doc

artgibson commented on src/locales/en.json:
```diff
@@ -72,6 +72,7 @@
   },
   "errors": {
     "description": "An error has occurred. Please reload the app.",
+    "maxed_submission_requests": "You have reached the max number of submissions, please wait 15 minutes to resubmit.",
```
15 to 5 minutes

artgibson reviewed (changes requested):
@lundjrl a unit test is currently failing.

lundjrl commented:
@artgibson I added a test case for the Callback form changes.

lundjrl committed:
parent 6488999cabeb8bc801437cdcdd349ad9bfd455af
author lundjrl <jrlbidamin@gmail.com> 1618490627 -0400
committer lundjrl <jrlbidamin@gmail.com> 1618490923 -0400

parent 6488999cabeb8bc801437cdcdd349ad9bfd455af
author lundjrl <jrlbidamin@gmail.com> 1618490627 -0400
committer lundjrl <jrlbidamin@gmail.com> 1618490627 -0400

GAEN-645 Adjusted Callback form to allow submissions every 5 minutes.

Update Form.tsx

Update Form.tsx

Changed name of value that is being saved to async storage

Update Form.tsx

Fixed error typo between 'error.' and 'errors.'

artgibson commented:
lgtm
